### ISSUE TITLE
consider renaming repo/prj to "vue-java" of "java-vue"

### ISSUE THREAD

**Issue by janblok** [Axellience/vue-gwt#27]
"gwt" does makes most developers sad, it sounds old and outdated.
don't get me wrong I like gwt a lot, but it just does not sell...to developers/management
basically you are using only gwtc which could be j2cl in the future.
both are transpilers to get from java to js
the project is not using most of the stuff gwt is known for anyways like the components
please drop "gwt" from the title

**Comment by Xaelis** [comment]
Hi Jan,

I agree! Gwt has now an outdated image for most developers.
We are considering renaming it "Javue". What do you think?

**Comment by janblok** [comment]
From a SEO perspective it might be beter to have "java" and "vue" as separate words.

**Comment by ibaca** [comment]
You are right, GWT has pretty bad popularity, but this lib only works with GWT anyway and adding the "java" suffix if false bc this cannot be used in the JVM. You can go directly to vue-j2cl, but IMO if someone sees GWT and don't like it, he will not use the lib anyway if the lib is named vue-java. Hehe, I agree with you but I don't understand how are you going to handle this conversation with a developer/management explaining that you want to use vue-java without naming GWT in the whole conversation 😅.

**Comment by janblok** [comment]
I guess it quite explainable that the lib only uses gwtc transpiler from gwt alike typescript transpilers  these days... and not the less favorite pieces like widgets/RPC.
Thanks to Axellience for considering the rename.

**Comment by adrienbaron** [comment]
Yea it's true and in the next version (beta-7) it's even less dependent as it drops GWT generators for Annotations Processors.
So with that version any Java to JS transpiler with correct JDK emulation (like J2CL) should work (but it's not tested so nothing is sure).
We will see how to go about the rename, as it would make sense if we do this to rename packages etc.. and that would obviously cause a lot of breaking changes (even though search and replace all probably would be enough for migration).

**Comment by janblok** [comment]
breaking changes like renames are oke in beta stage :-)

**Comment by realityforge** [comment]
Have you considered ‘vue4j’ - the project I am working on is ‘react4j’ for exactly this reason:)

**Comment by adrienbaron** [comment]
@realityforge I'm not sure about vue4j, I never really liked the "4j" suffix, but that's personal taste ˆˆ
I'm considering vue-java (to be consistent with vue-typescript), but it's still up for debate in our company (vs JaVue).

**Comment by slavap** [comment]
I don’t see much benefits in renaming, GWT community is quite small, and removing word will change nothing, doubt that it will attract more developers. Personally I don’t like JaVue, sounds bad to me. What about j2vue, that way it will be consistent with j2cl and jsweet. Also vue-java is ok as well, though not a real truth.

**Comment by ibaca** [comment]
`vue-jsinterop`? this might be more j2cl friendly for all this ng wrappers. I still don't think it worth for this project, but maybe I'll start using it 😉.

**Comment by takyonfr** [comment]
Why not dejavue ? :D

**Comment by baldram** [comment]
> Yea it's true and the next version (beta-7) is even less dependent as it drops GWT generators for
> Annotations Processors.
> So with that version any Java to JS transpiler with correct JDK emulation (like J2CL) should work 
> (but it's not tested so nothing is sure).

@adrienbaron I see the topic is quite old, but maybe there are some news?
At the moment we have v1.0-beta-9 so it should be even closer. Is it possible to use VueGWT with J2CL? Did you consider to support Bazel? The google/j2cl is doing well and is frequently updated. 
Probably it's time to consider J2CL as a transpiler and replace the GWT one.

PS: If about the renaming. 
> I'm considering vue-java

I think that @ibaca proposition is good for both GWT and J2CL. 
If about vue-java please be careful to not share VAVR's fate. They were JavaSlang and Oracle forced them to not use "Java" word. The same Jave EE vs Jakarta EE. So I think there is no option for "vue-java".
It's hard to understand Oracle... ¯\_(ツ)_/¯

**Comment by adrienbaron** [comment]
Hi @baldram !

Vue GWT is actually polymorphic and support in theory both GWT2 and GWT3 (that uses J2CL as it's compiler). It doesn't use any specific GWT2 features in core, and the only specific GWT2 integrations (like the GWT Widgets integration) are in a separate optional `gwt2` dependency.
The only things that Vue GWT relies on for it's `Java <=> JS` bridge are `JsInterop` and `Elemental2` both of which are compatible with J2CL.

However GWT3 is still being worked on, mostly on the Maven and Gradle plugins to make the J2CL pipeline more developer friendly. @niloc132 actually created a few PRs to add better GWT3 support to Vue GWT that got merged, and the current develop was capable of building a demo project both in GWT2 and GWT3 in June: https://github.com/VueGWT/vue-gwt/tree/develop

I have to get around and find the time to release it though... But you can experiment by cloning that branch locally and running `mvn install` then depending on `1.0-beta-10-SNAPSHOT` :).

For Bazel support I'm not sure what would be required, basically as long as the annotation processor runs and rerun on changes to Java files, Vue GWT doesn't need anything special from the build system.

There are also some Gradle GWT plugin out there that supports Webpack/NPM and Vue GWT: https://github.com/ascendtech/gwt-gradle

As for the name J2CL should be released to the more general public as `GWT 3` (as by itself it doesn't do much and needs a whole pipeline around for developer friendliness), so the name should work even for that 👍
I would be curious if you manage to get a project working with Bazel + Vue GWT + J2CL though, that would be quite cool!

**Comment by baldram** [comment]
The combination of Gradle, Webpack and Vue GWT seems to be really exciting. Thanks for pointing there.
